api4/user_test.go

```diff
@@ -5882,6 +5889,94 @@ func TestThreadSocketEvents(t *testing.T) {
 		require.Truef(t, caught, "User should have received %s event", model.WebsocketEventThreadReadChanged)
 	})
 
+	_, resp, err = th.Client.UpdateThreadReadForUser(th.BasicUser.Id, th.BasicTeam.Id, rpost.Id, rpost.CreateAt)
+	require.NoError(t, err)
+	CheckOKStatus(t, resp)
+
+	t.Run("Listen for read event 2", func(t *testing.T) {
+		var caught bool
+		func() {
+			for {
+				select {
+				case ev := <-userWSClient.EventChannel:
+					if ev.EventType() == model.WebsocketEventThreadReadChanged {
+						caught = true
+
+						data := ev.GetData()
+						require.EqualValues(t, rpost.CreateAt, data["timestamp"])
+						require.EqualValues(t, float64(0), data["previous_unread_replies"])
+						require.EqualValues(t, float64(0), data["previous_unread_mentions"])
+						require.EqualValues(t, float64(1), data["unread_replies"])
+						require.EqualValues(t, float64(1), data["unread_mentions"])
+
+					}
+				case <-time.After(1 * time.Second):
+					return
+				}
+			}
+		}()
+
+		require.Truef(t, caught, "User should have received %s event", model.WebsocketEventThreadReadChanged)
+	})
+
+	// read the thread
+	_, resp, err = th.Client.UpdateThreadReadForUser(th.BasicUser.Id, th.BasicTeam.Id, rpost.Id, replyPost.CreateAt+1)
+	require.NoError(t, err)
+	CheckOKStatus(t, resp)
+
+	t.Run("Listen for thread updated event after create post", func(t *testing.T) {
+		testCases := []struct {
+			post        *model.Post
+			preReplies  float64
+			preMentions float64
+			replies     int64
+			mentions    int64
+		}{
+			{&model.Post{ChannelId: th.BasicChannel.Id, Message: "simple reply", UserId: th.BasicUser2.Id, RootId: rpost.Id}, 0, 0, 1, 0},
+			{&model.Post{ChannelId: th.BasicChannel.Id, Message: "mention reply 1 @" + th.BasicUser.Username, UserId: th.BasicUser2.Id, RootId: rpost.Id}, 1, 0, 2, 1},
+			{&model.Post{ChannelId: th.BasicChannel.Id, Message: "mention reply 2 @" + th.BasicUser.Username, UserId: th.BasicUser2.Id, RootId: rpost.Id}, 2, 1, 3, 2},
+			// the following case is a broken
+			// there are a couple of issues,
+			// the results that passes the test are 0, 2, 0, 2
+			// when they should be 3, 2, 0, 0
+			// which means 1. we are zeroing unread but not mentions for some reason.
+			// and 2. previous unread are zeroed when they shouldn't
+			// not an actual problem, (2.) since to post you have to view the thread first
+			// could be a problem if we have quick replies in the app though (reply from a notification)
+			// {&model.Post{ChannelId: th.BasicChannel.Id, Message: "self reply 2", UserId: th.BasicUser.Id, RootId: rpost.Id}, 3, 2, 0, 0},
```

Comment:
Shouldn't the post here be created by BasicUser**2**?
i.e.
```suggestion
			// {&model.Post{ChannelId: th.BasicChannel.Id, Message: "self reply 2", UserId: th.BasicUser2.Id, RootId: rpost.Id}, 3, 2, 0, 0},
```

Then the expected result will be (3, 2, 4, 2). 
I don't understand how we're supposed to get (3, 2, 0, 0) in either case.

Reply:
since it's the current user posting the unread should not increment. 
so it's either 3, 2, 0, 0 or 3, 2, 3, 2(?)